src/main.c

```diff
@@ -182,8 +182,8 @@ int surjectionproof_parse(size_t *n_inputs, unsigned char *used_inputs, unsigned
   secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_ALL);
   secp256k1_surjectionproof proof;
   memcpy(&(proof.n_inputs), n_inputs, sizeof(proof.n_inputs));
-  memcpy(proof.used_inputs, used_inputs, 32);
-  memcpy(proof.data, data, 8224);
+  memcpy(&(proof.used_inputs), used_inputs, 32);
+  memcpy(&(proof.data), data, 8224);
```

Comment:
I don't get this magic bumber 🤔

Reply:
https://github.com/ElementsProject/secp256k1-zkp/blob/secp256k1-zkp/include/secp256k1_surjectionproof.h#L50
those are the fixed length of the surjection proof fields.